packages/discord.js/typings/index.d.ts

```diff
@@ -3277,6 +3254,37 @@ export class Webhook extends WebhookMixin() {
   public send(options: string | MessagePayload | WebhookMessageCreateOptions): Promise<Message>;
 }
 
+export class IncomingWebhook extends BaseWebhook {
+  public type: WebhookType.Incoming;
+  public applicationId: Snowflake | null;
+  public sourceGuild: null;
+  public sourceChannel: null;
+  public token: string;
+}
+
+export class UserCreatedWebhook extends IncomingWebhook {
+  public applicationId: null;
+  public owner: User | APIUser;
+}
+
+export class ApplicationCreatedWebhook extends BaseWebhook {
+  public type: WebhookType.Application;
+  public applicationId: Snowflake;
+  public sourceGuild: null;
+  public sourceChannel: null;
+  public token: string;
+}
+
+export class ChannelFollowerWebhook extends BaseWebhook {
+  public type: WebhookType.ChannelFollower;
+  public applicationId: null;
+  public sourceGuild: Guild | APIPartialGuild;
+  public sourceChannel: NewsChannel | APIPartialChannel;
+  public token: null;
+}
+
+export type Webhook = UserCreatedWebhook | ApplicationCreatedWebhook | IncomingWebhook | ChannelFollowerWebhook;
```

Comment:
here we export `Webhook` as a `type` and not as a `class` so the user cannot import it and use it as a class.